src/cargopacket.h

```diff
@@ -39,15 +39,27 @@ extern SaveLoadTable GetCargoPacketDesc();
  */
 struct CargoPacket : CargoPacketPool::PoolItem<&_cargopacket_pool> {
 private:
+	/* A mathematical vector from (0,0). */
+	struct Vector {
+		int16_t x;
+		int16_t y;
+	};
+
 	uint16_t count{0}; ///< The amount of cargo in this packet.
 	uint16_t periods_in_transit{0}; ///< Amount of cargo aging periods this packet has been in transit.
 
 	Money feeder_share{0}; ///< Value of feeder pickup to be paid for on delivery of cargo.
 
 	TileIndex source_xy{INVALID_TILE}; ///< The origin of the cargo.
+	Vector distance_traveled{0, 0}; ///< A vector (from 0,0) of how far the cargo got actually transported in a vehicle. Is a vector when in station. Is a temporary variable when in vehicle.
```

Comment:
it's not really a distance though. At least, I'd expect "distance travelled" to be a total length of the path taken. Also, British spelling seems to be "travelled".
P.S. "Displacement" seems to be a good word for this.

Reply:
Bikeshedding, but yeah. "distance" is only one of the two pieces of information in a vector. Going to drop "distance", so just "travelled" (and sure, as we write colour, I will write travelled).